File: java/src/main/java/it/giannibombelli/wsc2026/common/domain/identity/EntityId.java
package it.giannibombelli.wsc2026.common.domain.identity;

import java.util.UUID;
import java.util.function.Function;

public interface EntityId {
    UUID value();

    static <ID extends EntityId> ID generate(Function<UUID, ID> factory) {
        return factory.apply(UUID.randomUUID());
    }
}

File: typescript/src/common/domain/identity/entityId.ts
import { Uuid } from '@/common/domain/primitive/uuid.js';

export interface EntityId {
  readonly value: Uuid;
}

export function generateEntityId(): Uuid {
  return Uuid.generate();
}

export function generateId<T>(factory: (id: Uuid) => T): T {
  return factory(generateEntityId());
}
